src/main/java/com/faforever/server/geoip/GeoIpService.java

```diff
@@ -63,10 +64,30 @@ public void updateDatabaseFile() throws IOException {
   }
 
   public Optional<String> lookupCountryCode(InetAddress inetAddress) {
+    Optional<CityResponse> city = lookupCity(inetAddress);
+    return city.isPresent() ? Optional.of(city.get().getCountry().getIsoCode()) : Optional.empty();
```

Comment:
```
return lookupCity(inetAddress)
    .map(city -> city.getCountry().getIsoCode());
```

IntelliJ actually helps you half the way:
![image](https://cloud.githubusercontent.com/assets/1864183/24667475/e0af9826-1963-11e7-80b4-047591679eca.png)